core/templates/services/user-backend-api.service.ts

```diff
@@ -88,6 +96,9 @@ export class UserBackendApiService {
 
   private SITE_LANGUAGE_URL = '/save_site_language';
 
+  // Map from username to profile image.
+  private profileImageCache: Map<string, Blob> = new Map();
```

Comment:
I'm caching all profiles that the user views from `/profile/<username>` though we don't need to do this!

Reply:
Yeah I don't think it is needed.